qt/scientific_interfaces/Muon/MuonAnalysisFitDataPresenter.cpp

```diff
@@ -303,7 +303,15 @@ std::vector<std::string> MuonAnalysisFitDataPresenter::generateWorkspaceNames(
   const auto periods = m_dataSelector->getPeriodSelections();
 
   Muon::DatasetParams params;
-  const std::string instRuns = instrument + runString;
+  std::string runNumber = runString;
+  auto index = runString.find(instrument);
+  if (index != std::string::npos) {
+    // trim path
+    runNumber = runString.substr(index + instrument.size());
+    // trim extension
+    runNumber = runNumber.substr(0, runNumber.find_first_of("."));
+  }
+  const std::string instRuns = instrument + runNumber;
   std::vector<int> selectedRuns;
   MuonAnalysisHelper::parseRunLabel(instRuns, params.instrument, selectedRuns);
```

Comment:
Should we wrap this in a `try/catch`? If not it would still be possible for an exception generated by `parseRunLabel` to escape.

Reply:
I think that catches in parseRunLabel should catch everything. I could add another one here for safety